### ISSUE TITLE
Question 4 - the definition of a closure is incorrect

### ISSUE THREAD

**Issue by vibecode-zz** [ganqqwerty/123-Essential-JavaScript-Interview-Questions#90]
I believe the definition of a closure as "a function defined inside another function..." is not particularly correct. A closure is a function + lexical environment it's bound to. It doesn't necessary require an outer function.

See also https://stackoverflow.com/questions/50529400/is-the-definition-of-javascript-closures-on-mdn-wrong

**Comment by ganqqwerty** [comment]
You're right, I like Kyle Simpson definition much more after I read his books. May I ask you to create a pullrequest?
